### PR TITLE
fix: hybrid matview count to prevent boundary bucket over-counting in paginated log search

### LINKED ISSUE #5329: [Bug]: `/api/logs` returns an incorrect `total_count` for time ranges of 24 hours or longer
### Prerequisites

- [x] I have searched existing issues and discussions to avoid duplicates
- [x] I am using the latest version (or have tested against main/nightly)

### Description

When PostgreSQL LogStore is used, GET `/api/logs` may return a `pagination.total_count` that does not match the logs accessible through pagination.

Log details are always queried from the raw logs table using the exact requested time range:
```
timestamp >= start_time
AND timestamp <= end_time
```
For materialized-view-eligible ranges of at least 24 hours, however, `total_count` switches to the hourly `mv_logs_hourly` materialized view:
```
hour >= date_trunc('hour', start_time)
AND hour <= end_time
```
Because each hour row represents a complete hourly bucket, this expands both boundaries:

- `start_time` is rounded down to the beginning of its hour, including logs before the requested start.
- The bucket containing `end_time` is included in full, including logs after the requested end.
- If `end_time` is exactly on an hour boundary, the complete hour beginning at `end_time` is still included because the predicate uses <=.

Consequently, `total_count` may include logs outside the requested interval even though those logs cannot appear in any page of the detail response.

### Steps to reproduce

1. Use PostgreSQL LogStore with `mv_logs_hourly` ready.
2. Create logs inside and outside the boundary portions of a 24-hour interval.
3. Call `/api/logs` with a range of at least 24 hours and non-hour-aligned boundaries.
4. Compare `pagination.total_count` with the number of rows accessible through pagination.

### Expected behavior

`total_count` matches the number of logs within the requested time range.

### Actual behavior

`total_count` includes logs outside the requested time range because both boundaries are expanded to hourly buckets.

### Affected area(s)

Framework

### Version

latest

### Environment

```text

```

### Relevant logs/output

```shell

```

### Regression?

_No response_

### Severity

Medium (some functionality impaired)

### PR DESCRIPTION
**Pull request by impoiler**
## Summary

Fixes a pagination `total_count` bug where matview-eligible windows (≥24h) returned an inflated count because `mv_logs_hourly` predicates on `hour` rounded both time boundaries out to full hour buckets, counting logs that the exact-range row list could never page to. Closes #5329.

## Changes

- Replaced the single `mv_logs_hourly` sum in `getCountFromMatView` with a hybrid approach: only buckets fully contained within `[StartTime, EndTime]` are summed from the matview; the partial boundary buckets (at most one on each side) are counted directly from the raw `logs` table using a timestamp-indexed scan over ≤2 hours of rows.
- Introduced `countRawTerminal` to count raw log rows restricted to terminal statuses when no status filter is present, ensuring both halves of the hybrid count cover the same population as `mv_logs_hourly`.
- All bucket-grid arithmetic is performed in SQL via `date_trunc('hour', timestamptz)` rather than Go-side `time.Truncate`, so servers with fractional-hour timezone offsets (e.g. `Asia/Kolkata`, +05:30) produce correct bucket boundaries without misalignment.
- Added `TestSearchLogsMatViewCountMatchesRawRange` to assert that `total_count` matches the number of terminal logs strictly within `[StartTime, EndTime]` across mid-hour, hour-aligned-start, and hour-aligned-end boundary cases.

## Type of change

- [x] Bug fix

## Affected areas

- [x] Core (Go)

## How to test

```sh
go test ./framework/logstore/... -run TestSearchLogsMatViewCountMatchesRawRange -v
```

The test inserts logs at known timestamps straddling a 25h15m window, refreshes the matviews, forces the matview count path, and asserts that `total_count` equals exactly the number of terminal logs inside `[start, end]` for three boundary configurations (mid-hour boundaries, hour-aligned end, hour-aligned start).

## Breaking changes

- [x] No

## Related issues

Closes #5329

## Security considerations

None.

## Checklist

- [ ] I read `docs/contributing/README.md` and followed the guidelines
- [x] I added/updated tests where appropriate
- [ ] I updated documentation where needed
- [x] I verified builds succeed (Go and UI)
- [ ] I verified the CI pipeline passes locally if applicable

### CODE REVIEW

**Comment by impoiler** [comment]
* **#5507** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5507?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5486** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5486?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5486?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#5483** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5483?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5482** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/5482?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: framework/logstore/matviews.go [hunk 1]
```diff
@@ -1006,40 +1235,122 @@ func (s *RDBLogStore) getStatsFromMatView(ctx context.Context, filters SearchFil
...
+			SUM(count) AS total,
+			SUM(success_count) AS success,
+			SUM(error_count) AS error_count,
+			SUM(cancelled_count) AS cancelled_count
+		`, bucketSizeSeconds, bucketSizeSeconds)).
+			Group("bucket_timestamp").
+			Find(&results).Error; err != nil {
+			return nil, err
+		}
+		merge(results)
```
[the remark this answers is not in the document]
Comment (akshaydeo) [reply framework/logstore/matviews.go:1316]: Fixed - all 13 matview histogram gates in framework/logstore/rdb.go now require `bucketSizeSeconds%3600 == 0` in addition to `>= 3600`, so non-hour-aligned bucket sizes (e.g. 5400) fall back to the raw-table path with exact timestamp predicates. See updated code.

**Comment by akshaydeo** [comment]
### Merge activity

* **Jul 23, 10:59 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/maximhq/bifrost/5486).
* **Jul 23, 10:59 PM UTC**: @akshaydeo merged this pull request with [Graphite](https://app.graphite.com/github/pr/maximhq/bifrost/5486).
